SQL/0000-00-00-schema.sql

```diff
@@ -615,73 +615,63 @@ CREATE TABLE `files_qcstatus` (
       FOREIGN KEY (`FileID`) REFERENCES `files` (`FileID`)
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8;
 
-CREATE TABLE `mri_protocol_group` (
-    `MriProtocolGroupID`   INT(4) UNSIGNED NOT NULL AUTO_INCREMENT,
-    `Name`                 VARCHAR(255)    NOT NULL UNIQUE,
-    PRIMARY KEY (`MriProtocolGroupID`)
+CREATE TABLE `scan_type_parameter_group` (
+    `ScanTypeParameterGroupID`   INT(4) UNSIGNED NOT NULL AUTO_INCREMENT,
+    `ScanTypeParameterGroupName` VARCHAR(255)    NOT NULL UNIQUE,
+    PRIMARY KEY (ScanTypeParameterGroupID)
 ) ENGINE = InnoDB  DEFAULT CHARSET=utf8mb4;
 
-INSERT INTO `mri_protocol_group` (`Name`) VALUES('Default MRI protocol group');
-
-CREATE TABLE `mri_protocol` (
-  `ID` int(11) unsigned NOT NULL auto_increment,
-  `CenterID` integer unsigned DEFAULT NULL,
-  `ScannerID` int(10) unsigned DEFAULT NULL,
-  `Scan_type` int(10) unsigned NOT NULL default '0',
-  `TR_min` DECIMAL(10,4) DEFAULT NULL,
-  `TR_max` DECIMAL(10,4) DEFAULT NULL,
-  `TE_min` DECIMAL(10,4) DEFAULT NULL,
-  `TE_max` DECIMAL(10,4) DEFAULT NULL,
-  `TI_min` DECIMAL(10,4) DEFAULT NULL,
-  `TI_max` DECIMAL(10,4) DEFAULT NULL,
-  `slice_thickness_min` DECIMAL(9,4) DEFAULT NULL,
-  `slice_thickness_max` DECIMAL(9,4) DEFAULT NULL,
-  `xspace_min` int(4) DEFAULT NULL,
-  `xspace_max` int(4) DEFAULT NULL,
-  `yspace_min` int(4) DEFAULT NULL,
-  `yspace_max` int(4) DEFAULT NULL,
-  `zspace_min` int(4) DEFAULT NULL,
-  `zspace_max` int(4) DEFAULT NULL,
-  `xstep_min` DECIMAL(9,4) DEFAULT NULL,
-  `xstep_max` DECIMAL(9,4) DEFAULT NULL,
-  `ystep_min` DECIMAL(9,4) DEFAULT NULL,
-  `ystep_max` DECIMAL(9,4) DEFAULT NULL,
-  `zstep_min` DECIMAL(9,4) DEFAULT NULL,
-  `zstep_max` DECIMAL(9,4) DEFAULT NULL,
-  `time_min` int(4) DEFAULT NULL,
-  `time_max` int(4) DEFAULT NULL,
-  `image_type` varchar(255) default NULL,
-  `series_description_regex` varchar(255) default NULL,
-  `MriProtocolGroupID` INT(4) UNSIGNED NOT NULL,
-  PRIMARY KEY  (`ID`),
-  KEY `FK_mri_protocol_1` (`ScannerID`),
-  CONSTRAINT `FK_mri_protocol_1` FOREIGN KEY (`ScannerID`) REFERENCES `mri_scanner` (`ID`),
-  CONSTRAINT `FK_mri_protocol_2` FOREIGN KEY (`CenterID`) REFERENCES `psc` (`CenterID`),
-  CONSTRAINT `FK_mri_protocol_group_ID_1` FOREIGN KEY (`MriProtocolGroupID`) REFERENCES `mri_protocol_group` (`MriProtocolGroupID`)
+INSERT INTO scan_type_parameter_group (ScanTypeParameterGroupName) VALUES('Default MRI protocol group');
+
+CREATE TABLE `scan_type_parameter` (
+  `ScanTypeParameterID`      INT(11) UNSIGNED NOT NULL auto_increment,
+  `ScanTypeParameterGroupID` INT(4)  UNSIGNED NOT NULL,
+  `ScanType`   INT(10) UNSIGNED NOT NULL,
+  `HeaderName` VARCHAR(255)  NOT NULL,
+  `ValidMin`   DECIMAL(10,4) DEFAULT NULL,
```

Comment:
Is there ever any chance of there being multiple ranges that need to be supported for the same scan type? This will only support 1

Reply:
Would it? Nothing prevent the user to insert:
```
Scan_type    ValidMin  ValidMax.  HeaderName
T1           2000      2010       repetition_time
T1           2020      2030       repetition_time
```

The same way that a user can enter multiple row for a scan type in the current `mri_protocol` table. Maybe I am missing something?

Reply:
we discussed it during our imaging call. As long as the table is properly read by the imaging scripts, that should not be an issue to specify multiple ranges. Will resolve the discussion.